### ISSUE TITLE
Implement SL Training pipeline

### ISSUE THREAD

**Issue by JacquesDelfrate** [HeartWise-AI/DeepECG_Docker#54]
# Context
Currently, the `deepecg_docker` pipeline is utilized solely for external validation. With an upcoming submission and publication on the horizon, there is a need to implement a training and finetuning pipeline for both **Supervised Learning (SL)** and **Self-Supervised Learning (SSL)** approaches. 

This issue focuses on the implementation of the **SL model** pipeline. While we have a generic dataloader in place, we need to:

- Add a registry system to handle SL and SSL approaches including their respective loss functions and models 
- Develop a dedicated model class, as the existing setup only includes a wrapper for loading `.pt` files.
- Implement a couples of tools and system to improve maintainability of the `deepecg_docker` code base

The proposed solution is detailed in the section below. 

# Solution
- [ ] Implement registry system
- [ ] Upgrade arg parser system with training parameters
- [ ] Add config dataclass to handle the parameters of the different projects and approaches
- [ ] Provide with EfficientNet architecture or pre-trained model from pytorch
- [ ] Implement a training loop
- [ ] Implement validation loop
- [ ] Add loss(es) 
- [ ] Implement a logging system
- [ ] Implement wandb logger class for sweep (optional)
- [ ] Add training call to the run_pipeline.sh bash script
- [ ] Test
